### PR TITLE
Add Programs and Campaigns to seeds

### LINKED ISSUE #122: Add Programs and Campaigns to seeds
- We should have at least two programs with at least one campaign each
- Programs should have items
- Campaigns should have a few tasks, and maybe some riders assigned to them
- These should be idempotent like the current seeds
- It would be nice if the campaign's date is the day you run the seeds file.

### PR DESCRIPTION
**Pull request by mariusbutuc**
Screen Shot 2022-06-08 at 11 39 49 AM

Resolves #122.

### CODE REVIEW

File: lib/bike_brigade/delivery.ex [hunk 1]
```diff
@@ -677,10 +677,21 @@ defmodule BikeBrigade.Delivery do
...
+  def list_items(program_id \\ nil) do
     query = from i in Item, order_by: i.name
 
+    query =
+      if program_id do
+        query
+        |> where([i], i.program_id == ^program_id)
+      else
+        query
+      end
```
Comment (mariusbutuc) [inline lib/bike_brigade/delivery.ex:693]: Do we prefer to filter down by program id here, or rather leave it up to the caller, similar to [how campaigns are filtered below](https://github.com/bikebrigade/dispatch/pull/125/files#diff-c47a1aff9bf95e01794823e9077bbcf6c1b6321beadf2ace8ee9496415766c88R58)?

File: priv/repo/seeds/seeder.ex [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  def rider_for_campaign(campaign) do
+    rider =
+      Riders.list_riders()
+      |> Enum.random()
+
+    # validate_required([:campaign_id, :rider_id, :rider_capacity, :enter_building, :token])
+    Delivery.create_campaign_rider(%{
+      campaign_id: campaign.id,
+      rider_id: rider.id
+    })
```
Comment (mariusbutuc) [inline priv/repo/seeds/seeder.ex:44]: Should the other fields be required for seeded data as well?

File: priv/repo/seeds/seeder.ex [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+  def campaign() do
+    current_week =
+      LocalizedDateTime.today()
+      |> Date.beginning_of_week()
+
+    {:ok, campaign} =
+      Delivery.create_campaign(%{
+        # key :name not found in: %BikeBrigade.Delivery.Campaign{…}
+        name: Faker.Company.name(),
```
[the remark this answers is not in the document]
Comment (mariusbutuc) [reply priv/repo/seeds/seeder.ex:26]: Oops, the issue stems from here: https://github.com/bikebrigade/dispatch/blob/eb622fe4f6dab7c917d678d3d7a322a01f97da44/lib/bike_brigade_web/views/campaign_helpers.ex#L98-L106 especially line 104, which tells me the `campaign.program` relation is not set up correctly… debugging further.

File: priv/repo/seeds/seeder.ex (hunk 2 above)
Comment (mveytsman) [reply priv/repo/seeds/seeder.ex:43]: I think if the fields are not there this would fail right?

Comment (mveytsman) [reply priv/repo/seeds/seeder.ex:43]: nevermind they have defaults :)

File: priv/repo/seeds/seeder.ex (hunk 3 above)
[the remark this answers is not in the document]
Comment (mariusbutuc) [reply priv/repo/seeds/seeder.ex]: Resolved in #126.

File: priv/repo/seeds/seeder.ex [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+
+    rider
+  end
+
+  def task_for_campaign(campaign) do
+    {:ok, %Delivery.Task{} = task} =
+      campaign
+      |> Delivery.create_task_for_campaign(%{
+        dropoff_name: Faker.Person.first_name(),
+        dropoff_location: Toronto.random_location()
```
Comment (mariusbutuc) [inline priv/repo/seeds/seeder.ex:88]: In order to create a task (for a campaign) it seems that the drop-off name and location are required. Does this match "real life"? 

If so, thoughts on adding a `validate_required/3` for them to shorten the feedback loop and communicate this detail?

**Review by mveytsman — APPROVED** [review]
:shipit:

**Pull request merged**